packages/calypso-e2e/src/lib/blocks/pricing-table-block.ts

```diff
@@ -0,0 +1,56 @@
+import { Frame, Page } from 'playwright';
+import { BaseBlock } from '../base-block';
+
+const selectors = {
+	block: '.wp-block-coblocks-pricing-table',
+	pricing: '.wp-block-coblocks-pricing-table-item__amount',
+	gutterControl: 'div[aria-label="Editor settings"] div[aria-label="Gutter"]',
+};
+
+/**
+ * Represents the Pricing Table coblock.
+ */
+export class PricingTableBlock extends BaseBlock {
+	/**
+	 * Enters the price to the side chosen.
+	 *
+	 * @param {'left'|'right'} side Left or right side of the block.
+	 * @param {string} price Price to be entered.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async enterPrice( side: 'left' | 'right', price: string | number ): Promise< void > {
+		const index = side === 'left' ? 1 : 2;
+		const priceHandler = await this.block.waitForSelector(
+			`:nth-match(${ selectors.pricing }, ${ index })`
+		);
+		await priceHandler.fill( price.toString() );
+	}
+
+	/**
+	 * Given an appropriate value, select the gutter value on the settings sidebar for Pricing Table block.
+	 *
+	 * @param {string} value Value to set the gutter to.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async setGutter( value: 'None' | 'Small' | 'Medium' | 'Large' | 'Huge' ): Promise< boolean > {
+		const frame = ( await this.block.ownerFrame() ) as Frame;
+
+		const selector = `${ selectors.gutterControl } button[aria-label="${ value }"]`;
+		await frame.click( selector );
+		return await frame.$eval(
+			selector,
+			( element ) => element.getAttribute( 'aria-pressed' ) === 'true'
+		);
```

Comment:
The `$eval` call is not a `wait`-type call so it's possible to get a flaky result here. Do you think it would be better to  make this helper self-validate instead?

```suggestion
		const targetButton = await frame.waitForSelector( selector );
		await targetButton.click();
		await page.waitForFunction( ( element: any ) => element.ariaPressed === 'true', elementHandle ),
```

Reply:
Updated this call to be the following:

```
		await frame.click( selector );
		const elementHandle = await frame.waitForSelector( selector );
		await frame.waitForFunction(
			( element: any ) => element.ariaPressed === 'true',
			elementHandle
		);
```